contracts/HenkakuBadge.sol

```diff
@@ -39,26 +56,24 @@ contract HenkakuBadge is ERC1155, Ownable {
         uint256 _tokenId,
         bool _mintable,
         string memory _tokenURI
-    ) public onlyOwner {
-        require(
-            _tokenId > 0 && _tokenId <= _tokenIds.current(),
-            "Badge Not Exists"
-        );
+    ) public onlyOwner onlyExistBadge(_tokenId) {
         badges[_tokenId].mintable = _mintable;
         badges[_tokenId].tokenURI = _tokenURI;
         emit UpdateBadge(_tokenId, _mintable);
     }
 
-    function mint(uint256 _id, uint256 _amount) public {
-        // require for all badge attribute exists
-        // require amount is gte badge's amount to be piad
-        // require mintable is true
-        //  _mint(
-        //     address to,
-        //     uint256 id,
-        //     uint256 amount,
-        //     bytes memory data
-        // )
+    function mint(uint256 _tokenId) public onlyExistBadge(_tokenId) {
+        require(
+            erc20.balanceOf(msg.sender) >= badges[_tokenId].amount,
```

Comment:
I'm confused because the kamon one is "price" and the budge is "amount", is this an intentional distinction?
https://github.com/henkaku-center/kamon-nft/blob/main/contracts/kamonNFT.sol#L15

Reply:
hmmm, I didn't think of kamon actually.
amount is better than price.
since it is not actual price, just amount of the tokens.
so I'd prefer to use amount than price

Reply:
How about something like `amount_needed` or `erc20_cost`?

Just `amount` can be interpreted in different ways.

I like that `price` tells us that it is an amount that needs to be paid, but I agree about not using the word "price".

Reply:
amount is very common in cryptos. so I will use amount